tools/autograd/derivatives.yaml

```diff
@@ -330,15 +330,28 @@
 
 # For clamp, gradient is not defined at the boundaries. But empirically it's helpful
 # to be able to get gradient on min and max, so we return the subgradient 1 for these cases.
+- name: clamp.Tensor(Tensor self, Tensor? min=None, Tensor? max=None) -> Tensor
+  self: clamp_backward(grad, self, min, max)
+  min: grad * (self < min).to(grad.dtype())
```

Comment:
Perhaps use in all these formulas `masked_fill_` as:
`grad.clone().masked_fill_(self >= min, 0.)`
This way, if `grad` has `NaNs` we do not propagate them.

Reply:
Seems reasonable, changed to `where` to avoid the clone.